gravitee-apim-console-webui/src/management/api/runtime-logs-v4/runtime-logs-messages/components/api-runtime-logs-message-connection-log/api-runtime-logs-message-connection-log.component.html

```diff
@@ -0,0 +1,161 @@
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<mat-expansion-panel data-testId="entrypoint-request" class="log" expanded="true" *ngIf="!!connectionLogDetail.entrypointRequest">
+  <mat-expansion-panel-header class="log__header">
+    <mat-panel-title class="log__header__title">Entrypoint Request</mat-panel-title>
+  </mat-expansion-panel-header>
+
+  <div class="log__data">
+    <div class="log__row">
+      <div class="log__row__cell">
+        <div class="log__row__property">URI:</div>
+        <div data-testId="uri-value" class="log__row__value">{{ connectionLogDetail?.entrypointRequest?.uri | lowercase }}</div>
+      </div>
+    </div>
+    <div class="log__row">
+      <div class="log__row__cell">
+        <div class="log__row__property">Method:</div>
+        <div
+          data-testId="method-value"
+          class="{{ 'log__row__value gio-method-badge-' + connectionLogDetail?.entrypointRequest?.method?.toLowerCase() }}"
+        >
+          {{ connectionLogDetail.entrypointRequest.method | lowercase }}
+        </div>
+      </div>
+    </div>
+    <div class="log__row" *ngIf="Object.keys(connectionLogDetail?.entrypointRequest?.headers).length > 0">
+      <div class="log__row__cell">
+        <div class="log__row__property">Headers:</div>
+      </div>
+      <div
+        data-testId="header-value"
+        class="log__row__sub-cell"
+        *ngFor="let header of connectionLogDetail?.entrypointRequest?.headers | keyvalue"
+      >
+        <div class="log__row__property">{{ header.key }}:</div>
+        <div class="log__row__value">{{ header.value }}</div>
+      </div>
+    </div>
+  </div>
+</mat-expansion-panel>
+<mat-expansion-panel data-testId="endpoint-request" class="log" expanded="true" *ngIf="!!connectionLogDetail.endpointRequest">
+  <mat-expansion-panel-header class="log__header">
+    <div class="log__header__title">Endpoint Request</div>
+  </mat-expansion-panel-header>
+
+  <div class="log__data">
+    <div class="log__row" *ngIf="connectionLogDetail?.endpointRequest?.uri != ''">
+      <div class="log__row__cell">
+        <div class="log__row__property">URI:</div>
+        <div data-testId="uri-value" class="log__row__value">{{ connectionLogDetail?.endpointRequest?.uri | lowercase }}</div>
+      </div>
+    </div>
+    <div class="log__row">
+      <div class="log__row__cell">
+        <div class="log__row__property">Method:</div>
+        <div
+          data-testId="method-value"
+          class="{{ 'log__row__value gio-method-badge-' + connectionLogDetail?.endpointRequest?.method?.toLowerCase() }}"
```

Comment:
Here the code suggest `connectionLogDetail?.endpointRequest?.method?.toLowerCase()` but finally it should never be or your class won't make sense. I don't know if we should add a `*ngIf` on the `<div class="log__row">` and not use all the `?`